test/e2e/framework/util.go

```diff
@@ -1528,13 +1528,21 @@ func podsRunning(c *client.Client, pods *api.PodList) []error {
 	// are running so non-running pods cause a timeout for this test.
 	By("ensuring each pod is running")
 	e := []error{}
+	error_chan := make(chan error)
+
 	for _, pod := range pods.Items {
-		// TODO: make waiting parallel.
-		err := WaitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		go func() {
+			error_chan <- WaitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
```

Comment:
does this work? I'm worried that if the loop runs quickly enough the value of `pod` may change before this function gets executed.